collectors/0/kafka_jmx.py

```diff
@@ -0,0 +1,64 @@
+#!/usr/bin/python
+
+import sys
+
+from collectors.lib import utils
+from collectors.lib.jmx_monitor import JmxMonitor
+
+MONITORED_MBEANS = ["kafka.", "",                       # All kafka metrics.
+                    "Threading", "Count|Time$",        # Number of threads and CPU time.
+                    "OperatingSystem", "OpenFile",     # Number of open files.
+                    "GarbageCollector", "Collection"]  # GC runs and time spent GC-ing.
+
+
+class KafkaJmxMonitor(JmxMonitor):
+    USER = "kafka"
+    PROCESS_NAME = "kafka.Kafka"
+    METRIC_PREFIX = "kafka"
+
+    def __init__(self, pid, cmd):
+        super(KafkaJmxMonitor, self).__init__(pid,  cmd, MONITORED_MBEANS, self.PROCESS_NAME)
+
+    # Override
+    def process_metric(self, timestamp, metric, value, mbean):
+        jmx_service = ""
+        metric, tags = self.group_metrics(metric)
+
+        # mbean is of the form "domain:key=value,...,foo=bar"
+        # some tags can have spaces, so we need to fix that.
+        mbean_domain, mbean_properties = mbean.rstrip().replace(" ", "_").split(":", 1)
```

Comment:
yeah, I believe I stopped short of pulling up these lines in because then we'd need to pass through the tags too...but that's not a particularly good reason. Let me look at doing that.